Monika After Story/game/styles.rpy

```diff
@@ -388,3 +390,146 @@ init -10 python in mas_ui:
             if data["return_value"] == data["true_value"]:
                 return selected_prompt
         return default_prompt
+
+    # Methods for twopane_scrollable_menu
+    def _twopane_menu_filter_events(ev, search_query, search_kws, only_pool, only_random, only_unseen, only_seen):
+        """
+        A method to use as the filter for events in the twopane menu
+
+        IN:
+            ev - event object
+            search_query - search query to filter by
+            search_kws - search_query splitted using spaces
+
+        OUT:
+            boolean whether or not the event pass the criteria
+        """
+        ev_prompt = ev.prompt.lower()
+        ev_label = ev.eventlabel.lower()
+        ev_cat_full = " ".join(map(str, ev.category)) if ev.category else ""
+
+        return (
+            ev_prompt != ev_label
+            and ev.unlocked
+            # and (ev.pool or ev.random)
+            and not ev.anyflags(store.EV_FLAG_HFNAS)
+            and ev.checkAffection(store.mas_curr_affection)
+            and (
+                (not only_pool or ev.pool)
+                and (not only_random or ev.random)
+                and (not only_unseen or ev.shown_count == 0)
+                and (not only_seen or ev.shown_count > 0)
+            )
+            and (
+                search_query in ev_prompt
+                or search_query in ev_label
+                or search_query in ev_cat_full
+                or all(
+                    [
+                        search_kw in ev_prompt
+                        or search_kw in ev_label
+                        or search_kw in ev_cat_full
+                        for search_kw in search_kws
+                    ]
+                )
+            )
+        )
+
+    def _twopane_menu_sort_events(ev, search_query, search_kws):
```

Comment:
Can you explain to me how this function works with `sort`'s key?

Reply:
This function itself acts as a sort key. I had to use a `lambda` with it because I need to pass args in. The function makes some comparisons and returns a list of booleans with the results. The more hits an event gets, the higher it will be in the final list. The order of booleans is important too: first items in `rv` have more weight (hence I put comparisons with `ev_prompt` and `ev_label` first, that's also why I iter `search_kws` thrice).

Reply:
Ok so the list is to enforce a weighting system. Why not have the function return a number instead, where let's say, the bigger the number the higher the match? Something like:
```python
weight = 0
if matches_high_priority_criteria:
    weight += 2**10
if matches_lesser_criteria:
    weight += 2**5
# ... etc
return weight
```

It's not that clear how a list of bools would affect sorting. It might work, but a weighted score is easier to understand and helps maintainability.

Reply:
Not sure if that'd have the same effect in sorting, but I can try.